controllers/metallb_controller.go

```diff
@@ -144,6 +146,22 @@ func (r *MetalLBReconciler) syncMetalLBResources(config *metallbv1beta1.MetalLB)
 		if err := controllerutil.SetControllerReference(config, obj, r.Scheme); err != nil {
 			return errors.Wrapf(err, "Failed to set controller reference to %s %s", obj.GetNamespace(), obj.GetName())
 		}
+		if obj.GetKind() == "DaemonSet" && len(config.Spec.SpeakerDaemonNodeSelector) > 0 {
```

Comment:
Can't we inject the selector directly in the template?

Reply:
my understanding it has to be read modified write so I convert the object to ds then update it and write it back

Reply:
Not necessarily. If you look at the images for example, they are replaced directly in the template.

Reply:
let me try it to be sure

Reply:
yeah it broke they are different objects hence the need to Conversion